### PR TITLE
Make pool_name configurable for service_domain_db

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses MIM-1451

Proposed changes include:
* Use global as a default pool
* Make this option configurable

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/66261) / small_tests / f856c17d1742ef8ab1e409c963547f6855f6418d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3205/66261/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3205/66261/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-08-04_07.14.46/lib.mongooseim.logs/run.2021-08-04_07.14.46/suite.log.html)

File: doc/developers-guide/domain_management.md [hunk 1]
```diff
@@ -101,6 +101,21 @@ Configure host-type first to delete such domains.
...
+```
+
+### `pool_name`
+
+Pool name to use for a database connection.
+
+* **Syntax:** atom
+* **Default:** `global`
+* **Example:** `pool_name = "global"`
+
```
Comment (chrzaszcz) [inline doc/developers-guide/domain_management.md:118]: The problem with this is that we don't have such thing as a "pool name". In the code it has the type `host_type_or_global()`, so I would suggest something like the following:

```suggestion
### `db_pool`

By default, this service uses the RDBMS connection pool configured with the scope `"global"`.
You can put a specific host type there to use the pool with the `"host"` or `"single_host"`scope for that particular host type. See [PUT A LINK HERE] for more information about the pool scopes.

* **Syntax:** string
* **Default:** `global`
* **Example:** `db_pool = "my_host_type"`
```

The goal here is to make the user aware of what this name means and how to set up the corresponding pool.

**Review by chrzaszcz — COMMENTED** [review]
This PR fixes the bug, but I think that the docs are difficult to understand for the user, see details in the comment.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/66316) / small_tests / 7da39d87c59dfe34836b9e907a5a9bac7fad6bbe
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3205/66316/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3205/66316/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-08-04_15.14.37/lib.mongooseim.logs/run.2021-08-04_15.14.38/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good!

**Review by vkatsuba — APPROVED** [review]

**Pull request merged**